src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/AuthorisationImpl.java

```diff
@@ -0,0 +1,99 @@
+
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.audio.repository.MediaRequestRepository;
+import uk.gov.hmcts.darts.authorisation.api.AuthorisationApi;
+import uk.gov.hmcts.darts.authorisation.component.Authorisation;
+import uk.gov.hmcts.darts.cases.repository.CaseRepository;
+import uk.gov.hmcts.darts.common.entity.CourthouseEntity;
+import uk.gov.hmcts.darts.common.enums.SecurityRoleEnum;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.repository.MediaRepository;
+import uk.gov.hmcts.darts.common.repository.TranscriptionRepository;
+
+import java.util.List;
+import java.util.Set;
+
+import static uk.gov.hmcts.darts.audio.exception.AudioApiError.MEDIA_NOT_FOUND;
+import static uk.gov.hmcts.darts.audio.exception.AudioRequestsApiError.MEDIA_REQUEST_NOT_FOUND;
+import static uk.gov.hmcts.darts.cases.exception.CaseApiError.CASE_NOT_FOUND;
+import static uk.gov.hmcts.darts.hearings.exception.HearingApiError.HEARING_NOT_FOUND;
+import static uk.gov.hmcts.darts.transcriptions.exception.TranscriptionApiError.TRANSCRIPTION_NOT_FOUND;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class AuthorisationImpl implements Authorisation {
+
+    private final CaseRepository caseRepository;
+    private final HearingRepository hearingRepository;
+    private final MediaRequestRepository mediaRequestRepository;
+    private final MediaRepository mediaRepository;
+    private final TranscriptionRepository transcriptionRepository;
+    private final AuthorisationApi authorisationApi;
+
+    @Override
+    public void authoriseByCaseId(Integer caseId, Set<SecurityRoleEnum> securityRoles) {
+        try {
+            final List<CourthouseEntity> courthouses = List.of(caseRepository.getReferenceById(caseId)
+                                                                   .getCourthouse());
+            authorisationApi.checkAuthorisation(courthouses, securityRoles);
+        } catch (EntityNotFoundException e) {
+            log.error("Unable to find Case-Courthouse for checkAuthorisation", e);
+            throw new DartsApiException(CASE_NOT_FOUND);
+        }
+    }
+
+    @Override
+    public void authoriseByHearingId(Integer hearingId, Set<SecurityRoleEnum> securityRoles) {
+        try {
+            final List<CourthouseEntity> courthouses = List.of(hearingRepository.getReferenceById(hearingId)
+                                                                   .getCourtroom().getCourthouse());
+            authorisationApi.checkAuthorisation(courthouses, securityRoles);
+        } catch (EntityNotFoundException e) {
+            log.error("Unable to find Hearing-Courtroom-Courthouse for checkAuthorisation", e);
+            throw new DartsApiException(HEARING_NOT_FOUND);
```

Comment:
Probably not,  but are we leaking information here? I'm thinking some kind of enumeration attack.  Should we be informing the client that the record in question does not exist?

Reply:
I had a similar thought myself - it was done here to keep the similar responses on the Cases / Hearing API which responds with these Not Found errors. So I did similar for the other 3 contexts that I couldn't navigate to courthouses from. I wanted to distinguish between the Bad Request / Not Found / Unauthorized client errors to avoid confusion.
@hemantasharma1129 advises to keep it for now.